dbms/src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -539,6 +612,9 @@ void InterpreterSelectQuery::executeImpl(Pipeline & pipeline, const BlockInputSt
             options.to_stage == QueryProcessingStage::WithMergeableState)
             throw Exception("Distributed on Distributed is not supported", ErrorCodes::NOT_IMPLEMENTED);
 
+        if (storage && expressions.filter_info && expressions.prewhere_info)
```

Comment:
Why we check it again? Forgot to remove?

Reply:
Nope, we know about `prewhere_info` only after call to `analyzeExpressions()` - since we have two different `if` branches of code we have to check two times.